File: go/2015/01/solution.go
package main

import (
	"aoc/utils"
	"fmt"
)

func main() {
	data := utils.Read()
	floor := 0
	firstBasementAt := 0
	for i, c := range data {
		if c == '(' {
			floor += 1
		} else {
			floor -= 1
		}

		if floor < 0 && firstBasementAt == 0 {
			firstBasementAt = i + 1
		}
	}

	fmt.Println(floor)
	fmt.Println(firstBasementAt)
}

File: python/2015/01/solution.py
import common

data = common.read_file()

floor = 0
first_basement_at = 0
for i in range(len(data)):
    if data[i] == '(':
        floor += 1
    else:
        floor -= 1

    if floor < 0 and first_basement_at == 0:
        first_basement_at = i+1

print(floor)
print(first_basement_at)
